code/gui/griddisplay.cpp

```diff
@@ -107,5 +140,30 @@ void GridDisplay::drawGrid() {
 
 void GridDisplay::hideGrid() {
     //TODO
-    //gridDisplayed = false;
+    //hide or disable qgraphicsview
+    //gridDisplayed = true;
+}
+
+void GridDisplay::selectRobotPosition(QString weightSelected) {
+    if (weightSelected == "Start Configuration") {
+        startPosSelected = true;
+        endPosSelected = false;
+    } else if (weightSelected == "End Configuration") {
+        startPosSelected = false;
+        endPosSelected = true;
+    } else {
+        startPosSelected = false;
+        endPosSelected = false;
```

Comment:
there's a lot of branching here, and it'd be relatively easy to simplify the expressions